scalalib/worker/src/mill/scalalib/worker/ZincWorkerImpl.scala

```diff
@@ -708,4 +685,36 @@ object ZincWorkerImpl {
       toAnalyze = List((List(start), deps(start).toList))
     ).reverse
   }
+
+  private class AutoCloseableCache[A, B <: AnyRef] extends Cache[A, B] {
```

Comment:
Why is this called `AutoCloseableCache`? Did you forget a `with AutoCloseable`?

Also, I'd prefer to have `Cache` defined before its derivatives.

Reply:
What I want to express is a cache that on `clear` also tries to find `AutoCloseable` values and `close` them. I don't what is the best name for it. Do you have suggestions? I will change the order.